wpimath/src/main/java/edu/wpi/first/math/estimator/KalmanFilter.java

```diff
@@ -199,7 +205,16 @@ public void predict(Matrix<Inputs, N1> u, double dtSeconds) {
   public void correct(Matrix<Inputs, N1> u, Matrix<Outputs, N1> y) {
     final var C = m_plant.getC();
     final var D = m_plant.getD();
+    final var I = Matrix.eye(States);
+    
     // x̂ₖ₊₁⁺ = x̂ₖ₊₁⁻ + K(y − (Cx̂ₖ₊₁⁻ + Duₖ₊₁))
     m_xHat = m_xHat.plus(m_K.times(y.minus(C.times(m_xHat).plus(D.times(u)))));
+
+    // S = CPCᵀ + R
+    var S = C.times(m_P).times(C.transpose()).plus(m_discR);
+
+    m_K = S.transpose().solve((C.times(m_P.transpose()))).transpose();
+
+    m_P = I.minus(m_K.times(C)).times(m_P);
```

Comment:
Please add the equation comments:
```java
    // K = (Sᵀ.solve(CPᵀ))ᵀ
```
```java
    // Pₖ₊₁⁺ = (I − Kₖ₊₁C)Pₖ₊₁⁻
```

Reply:
@calcmogul added comments in commit 88c70de.